src/services/sdk.ts

```diff
@@ -0,0 +1,33 @@
+import { Graph, HexString, Profile } from "../utilities/types";
+import * as sdk from "./fakesdk";
+
+export const getSocialIdentity = async (
+  walletAddress: HexString
+): Promise<HexString> => {
+  let socialAddress: HexString = await sdk.getSocialIdentityfromWalletAddress(
+    walletAddress
+  );
+  if (!socialAddress) {
+    socialAddress = await sdk.createSocialIdentityfromWalletAddress(
+      walletAddress
+    );
+  }
+  return socialAddress;
```

Comment:
The thinking was that it shouldn't ever be null. Of course depending on the way the SDK works this may need to be reworked. We first ask the SDK for the socialAddress, and if that's null we create one. In either case we should get back a social identity unless there is an error.